numpy/core/tests/test_scalar_methods.py

```diff
@@ -102,3 +102,20 @@ def test_roundtrip(self, ftype, frac_vals, exp_vals):
                 pytest.skip("longdouble too small on this platform")
 
             assert_equal(nf / df, f, "{}/{}".format(n, d))
+
+class TestBitCount:
+    # derived in part from the cpython test "test_bit_count"
+
+    @pytest.mark.parametrize("itype", np.sctypes['int']+np.sctypes['uint'])
+    def test_small(self, itype):
+        for a in range(max(np.iinfo(itype).min, 0), 128):
+            msg = f"Smoke test for {itype}.bit_count({a})"
+            assert itype(a).bit_count() == bin(a).count("1"), msg
+
+    def test_bit_count(self):
+        for exp in [10, 17, 63]:
+            a = 2**exp
+            assert np.uint64(a).bit_count() == 1
```

Comment:
We hit an overflow here. Will take a look.

Reply:
Fixed with https://github.com/numpy/numpy/pull/19355/files#diff-e1d48225a79233f1670054e229f9b4c596929600773aca6d333a72da72897052R224